pkg/gadgets/trace/dns/tracer/bpf/dns.c

```diff
@@ -204,6 +232,36 @@ output_dns_event(struct __sk_buff *skb, union dnsflags flags, __u32 name_len, __
 	int anaddrcount = load_addresses(skb, ancount, anoffset, event);
 	event->anaddrcount = anaddrcount;
 
+	// Calculate latency:
+	//
+	// Filter by mount_ns_id and pkt_type to track the latency from when a query is sent from a container
+	// and when a response to the query is received by that same container.
+	//
+	// * On DNS query sent from a container namespace (mount_ns_id != 0, qr == 0, and pkt_type == OUTGOING),
+	//   store the query timestamp in a map.
+	//
+	// * On DNS response received in the same container namespace (mount_ns_id != 0, qr == 1, and pkt_type == HOST)
+	//   retrieve/delete the query timestamp and set the latency field on the event.
+	//
+	// A garbage collection thread running in userspace periodically scans for keys with old timestamps
+	// to free space occupied by queries that never receive a response.
+	if (event->mount_ns_id) {
+		struct query_key_t query_key = {.mount_ns_id = event->mount_ns_id, .id = event->id};
+		if (event->qr == DNS_QR_QUERY && event->pkt_type == PACKET_OUTGOING) { // query with type PACKET_OUTGOING
+			struct query_ts_t query_ts = {.timestamp = event->timestamp};
+			bpf_map_update_elem(&query_map, &query_key, &query_ts, BPF_NOEXIST);
+		} else if (event->qr == DNS_QR_RESP && event->pkt_type == PACKET_HOST) { // response with type PACKET_HOST
```

Comment:
nit: These two comments are not needed anymore.

Reply:
thanks, removed them